src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -24,45 +20,27 @@
  */
 trait LockableTrait
 {
-    private ?LockInterface $lock = null;
-
     /**
      * Locks a command.
      */
     private function lock(?string $name = null, bool $blocking = false): bool
     {
-        if (!class_exists(SemaphoreStore::class)) {
-            throw new LogicException('To enable the locking feature you must install the symfony/lock component. Try running "composer require symfony/lock".');
-        }
-
-        if (null !== $this->lock) {
-            throw new LogicException('A lock is already in place.');
-        }
-
-        if (SemaphoreStore::isSupported()) {
-            $store = new SemaphoreStore();
-        } else {
-            $store = new FlockStore();
+        if (null === $this->lockHandler) {
+            throw new LogicException('Lock handler is not available. You probably forgot to call setLockHandler method on your command.');
         }
 
-        $this->lock = (new LockFactory($store))->createLock($name ?: $this->getName());
-        if (!$this->lock->acquire($blocking)) {
-            $this->lock = null;
-
-            return false;
-        }
-
-        return true;
+        return $this->lockHandler->lock($name ?? $this->getName(), $blocking);
```

Comment:
Wouldn't this a BC break ?

The lockableTrait will stop using Semaphore/Flock store but instead will use the lockFactory configured in the Symfony config.

(Even if I like the fact we can use another store)

Reply:
Hello !

Yes, you're right. So I made multiple changes, to simplify the PR:

 - left the LockableTrait untouched, so no BC break;
 - remove the LockHandler to manager the lock in the Command class, to keep things simpler